sql/item_strfunc.cc

```diff
@@ -1348,13 +1348,31 @@ bool Item_func_sformat::fix_length_and_dec()
 
   for (uint i=0 ; i < arg_count ; i++)
   {
-    char_length+= args[i]->max_char_length();
+    // scan the number in {:d} and add it to the char_length
+    size_t num= 0;
+    const char* str= args[i]->name.str;
+    char delimiter1[]= "{:";
+    char delimiter2[]= "}";
```

Comment:
A couple of problems without a full solution.
`{{` is an escaped `{` so should be processed ref: https://fmt.dev/latest/syntax.html

The width isn't always the next argument - after `:` - https://fmt.dev/latest/syntax.html#format-specification-mini-language

Reply:
I'd be looking for a solution in the fmt API. @vuvova any other ideas of a solution to this one?

Reply:
Yes, using fmt api is a better choice. I also think about regrex to extract the number. What do you think?

Reply:
Also I found this api could return the length of result https://fmt.dev/latest/api.html#_CPPv4IDpEN3fmt14formatted_sizeE6size_t13format_stringIDp1TEDpRR1T

Reply:
This is not easy. @grooverdan already commented about `{{`. You would basically need to have a complete frm format parser to get the length right. And we don't want to have a second fmt format parser.

`fmt::formatted_size()` needs the string and all arguments. To call it you need to use dummy values for all arguments, like, `0` for `INT_RESULT`, `""` for `STRING_RESULT`, etc. This is quite a trick, so make sure to avoid it when it's not needed, like, when one of the arguments is already blob, so the length is already at the maximum.

perhaps it'd be enough to always return `MAX_BLOB_WIDTH` here without trying to estimate the length more precisely.